packages/block-library/src/paragraph/editor.scss

```diff
@@ -15,3 +15,8 @@
 	min-height: $empty-paragraph-height / 2;
 	line-height: $editor-line-height;
 }
+
+// Overwrite the inline style to make the height collapse when the paragraph editable gets focus.
+.block-editor-block-list__block[data-type="core/paragraph"] .has-drop-cap:focus {
+	min-height: $empty-paragraph-height / 2 !important;
```

Comment:
It would be clearer to just not apply the inline style when focused, using the selection props/selectors.

Reply:
The isSelected prop gives an indication if the block is selected or not but the block can be selected and we may not have the rich text-focused e.g: we are typing on a block inspector field. As far as I know, there is no prop that tells us if the rich text is focused or not.